cli/src/cli_util.rs

```diff
@@ -2420,10 +2433,12 @@ pub struct GlobalArgs {
     /// Operation to load the repo at
     ///
     /// Operation to load the repo at. By default, Jujutsu loads the repo at the
-    /// most recent operation. You can use `--at-op=<operation ID>` to see what
-    /// the repo looked like at an earlier operation. For example `jj
-    /// --at-op=<operation ID> st` will show you what `jj st` would have
-    /// shown you when the given operation had just finished.
+    /// most recent operation. If there are concurrent operations, they will be
+    /// merged. You can use `--at-op=<operation ID>` to see what the repo looked
+    /// like at an earlier operation. For example `jj --at-op=<operation ID> st`
+    /// will show you what `jj st` would have shown you when the given operation
+    /// had just finished. `--at-op=@` is pretty much the same as the default
+    /// except that concurrent operations will never be merged.
```

Comment:
I find it a bit confusing that `--at-op=@` does not seem to imply `--ignore-working-copy`. Is this important, or can we make it more consistent by making it imply `--ignore-working-copy`?

If so, I'd delete this sentence and change one of the paragraphs below to:


> Using `--at-op` has two effects in addition to loading an earlier operation. First, the working copy is ignored, as if `--ignore-working-copy` was specified. Second, `jj` will *not* merge concurrent operations, if any, like it does by default.

Reply:
One  confusing implication (even if possibly rarely in play) would be that if `abc` happens to refer to `@`, then `--at-op=abc` will not imply `--ignore-working-copy`, but once another operation happens, the same option will imply `--ignore-working-copy`.

Reply:
I don't think `--at-op=@` should disable snapshot. And I personally think it's a bug that `--at-op=<head_id>` implies `--ignore-working-copy`.

The idea is that `--at-op=<no_head>` implies `--ignore-working-copy` because the working copy doesn't represent the specified operation.

Reply:
>  I personally think it's a bug that --at-op=<head_id> implies --ignore-working-copy.

I can understand this from a purity standpoint. I find the idea of "fixing" this bug quite confusing though; I can't tell how bizarre the resulting behavior of `jj log --at-op @-` would be. Presumably, the result would be:

```
@ merge
| \
|  o new_snapshot
|  |
o  | former_@
| /
o  former_@-
```

whereas the user would likely expect (if they expect the operation to snapshot at all)

```
@ new_snapshot
o former_@
o former_@-
```

I do not understand merging well enough to understand whether the two resulting states `@` are the same or not. If not, I imagine the difference would be quite confusing.

------------------

Assuming you see a solution to the above problem, I'd be OK with "fixing the bug". If we don't fix the bug, I'd prefer consistently ignoring the working copy always, except perhaps if there is some use-case where snapshotting it is helpful, for example because of the confusing situation that you can't tell whether `jj log --at-op abc` will ignore working copy without knowing what `abc` is.

All that said, I'll ultimately leave this up to you and Martin, you've been discussing this for a while and my understanding of these issues feels a bit superficial at the moment.

Reply:
Or would `jj log --at-op @-` just fail (if it `--at-op` does not imply `--ignore-working-copy`) because after the snapshot and the merge, `@-` would no longer refer to `former_@-` as the user expects, but refers to two operations?

Reply:
Maybe I don't follow your point, but `jj --at-op @-` fails if there are more than one  op heads. That's unchanged. This patch makes it even stricter, `jj --at-op @` now fails.

Oh, maybe you mean `jj --at-op @-` could do snapshot (if `@-` can be resolved to a single operation)? I don't think enabling snapshot makes sense in that situation (though it's technically possible if the working copy tree is identical to the `@-` one.)

What I meant is `jj --at-op @` and `jj --at-op <head-id>` (where `<head-id>` points to the current operation head) don't have to imply `--ignore-working-copy`.

Reply:
I guess I misunderstood you. My understanding was that you suggested that `--at-op anything` shouldn't imply `--ignore-working-copy`, but I see that was wrong.

I also thought that with your implementation, neither `--at-op @` nor `--at-op <head_id>` implied `--ignore-working-copy`. It sounds like what you were actually saying is that while `--at-op @` *does not* imply `--ignore-working-copy`, `--at-op <head_id>` *does*, and *that*'s what you thought was a bug. 

To me, it would seem that there is no good choice for `--at-op <head_id>` if `--at-op <another_id>` and `--at-op @` have different behavior. If we **had** to choose I'd agree with you that making it behave the same as `--at-op @` is probably better, but it doesn't seem good.

This is why I thought it'd be simpler to make `--at-op <anything>` imply `--ignore-working-copy`. The only downside I see is that it's hard to have the behavior of "do snapshot but don't merge concurrent heads", but I don't know when that would be desired. If there is a use-case, we could perhaps make `jj debug snapshot --at-op <anything>` not `--ignore-working-copy`, which would cover that and other weird cases (for testing snapshotting maybe? I'm unsure).

But again, I could easily be missing something important still.

Reply:
> while `--at-op @` _does not_ imply `--ignore-working-copy`, `--at-op <head_id>` _does_, and _that_'s what you thought was a bug.

Correct. It wouldn't practically matter, but seemed a bit odd.

> This is why I thought it'd be simpler to make `--at-op <anything>` imply `--ignore-working-copy`. The only downside I see is that it's hard to have the behavior of "do snapshot but don't merge concurrent heads",

I don't think this operation mode (= don't merge ops, do snapshot) is useful, but I don't think `--at-op=@` should disable snapshot either. These two flags are conceptually different. That said, I'm not strongly against your idea because the implementation will get simpler.

Updated the todo comment.